app/src/main/java/com/github/studydistractor/sdp/ui/CreateEventScreen.kt

```diff
@@ -0,0 +1,287 @@
+            )
+
+            PointsAwardedField(createActivityViewModel = createEventViewModel, uiState = uiState)
+            LateParticipationCheckbox(
+                createEventViewModel = createEventViewModel,
+                uiState = uiState
+            )
+
+            CreateActivityButton(
+                createActivityViewModel = createEventViewModel,
+                onActivityCreated = onEventCreated,
+                buttonText = "Create event"
+            )
+        }
+    }
+}
+
+/**
+ * Creates a date picker dialog
+ * @param context Context
+ * @param setDateTime Callback to be called when date and time is set
+ * @param year Year
+ * @param month Month
+ * @param dayOfMonth Day of month
+ * @param hour Hour
+ * @param minute Minute
+ * @return DatePickerDialog
+ */
+private fun dateTimePickerDialog(
+    context: Context,
+    setDateTime: (year: Int, month: Int, day: Int, h: Int, m: Int) -> Unit,
+    year: Int,
+    month: Int,
+    dayOfMonth: Int,
+    hour: Int,
+    minute: Int
+): DatePickerDialog {
+    return DatePickerDialog(
+        context,
+        { _, y, m, d ->
+            val timePicker = TimePickerDialog(
+                context,
+                { _, h, min -> setDateTime(y, m, d, h, min) },
+                hour,
+                minute,
+                true
+            )
+            timePicker.show()
+        },
+        year,
+        month,
+        dayOfMonth
+    )
+}
+
+/**
+ * Creates a text field to allow user to enter date and time
+ * @param valueText Text to be displayed in the text field
+ * @param onValueChange Callback to be called when text is changed
+ * @param labelText Label text
+ * @param onClickButton Callback to be called when button is clicked
+ * @return TextField
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CreateDateTimeField(
+    valueText: String,
+    onValueChange: (String) -> Unit,
+    labelText: String,
+    onClickButton: () -> Unit,
+) {
+    OutlinedTextField(
+        value = valueText,
+        onValueChange = { onValueChange(it) },
+        label = { Text(labelText) },
+        trailingIcon = {
+            Button(
+                onClick = { onClickButton() },
+                colors = ButtonDefaults.buttonColors(contentColor = Color.White),
+                modifier = Modifier
+                    .testTag(labelText + "Button")
+                    .padding(8.dp),
+                shape = MaterialTheme.shapes.small,
+            ) {
+                Icon(Icons.Filled.CalendarMonth, contentDescription = null)
+            }
+        },
+        singleLine = true,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(vertical = 8.dp)
+            .testTag(labelText + "Field")
+    )
+}
+
+/**
+ * Creates a checkbox to allow user to select if late participation is allowed
+ * @param createEventViewModel ViewModel
+ * @param uiState UI state
+ * @return Row
+ */
+@Composable
+fun LateParticipationCheckbox(
```

Comment:
Does this deserve its own composable? If so, it should probably be a component.

Reply:
It's a separate composable in an effort to keep the code a bit more readable, rather than having a giant composable with everything inside. I do not think it should be it's own composable, since it is specific to the `CreateEventScreen`.

Reply:
alright! this is far too subjective anyway.